plugins/modules/cloud/misc/terraform.py

```diff
@@ -444,9 +453,12 @@ def main():
     else:
         plan_file, needs_application, out, err, command = build_plan(command, project_path, variables_args, state_file,
                                                                      module.params.get('targets'), state, plan_file)
+        if state == 'present' and check_destroy and '- destroy' in out:
+            module.fail_json(msg="Aborting command because it would destroy some resources. "
+                                 "Consider switching the 'check_destroy' to false to suppress this error")
```

Comment:
It makes more sense to me to exit instead of fail here as the the user intended for the module to prevent application, but if it stays as fail it would be better to include `out` so the user can inspect which resource(s) would have been destroyed.

Reply:
> It makes more sense to me to exit instead of fail here as the the user intended for the module to prevent application, but if it stays as fail it would be better to include `out` so the user can inspect which resource(s) would have been destroyed.

I don't agree entirely with that statement. If there is no resource being destroyed in the application, it will not be prevented. So the end result, even with `check_destroy: yes` may be either a successful application of the plan, or a failure to do so. I still feel like it makes sense to have the two distinct endings. Maybe a normal exit with some flag indicating this case would do the trick, but... mixed feelings about it.

Reply:
The more important part to me is providing meaningful output about the affected terraform resources to the user. As long as the module is consistent about how it stops execution in this scenario I think the user can adjust. My only argument for exiting is I think the user would expect the task to be `changed` on successful application, `success; changed: false` when `check_destroy` has effect, and `failed` when something unexpected happened. But `success; changed: false` covers other scenarios as well so it could go either way.

Reply:
You got a point there, for sure. IMO the fact that some resource will be destroyed (when state=present) fits nicely the definition of "something unexpected happened".

I am thinking the likely scenarios (IMO) when this will be used are:
* The user wants to have a distinct process for handling these cases (when resources are destroyed) - maybe backing up the data elsewhere before doing it, although I would vouch for backing it up regardless of "destruction".
* The user wants the whole process to stop and manually take a look at it before running it automatically again.

I don't think it makes sense to return `success` in any of these cases.

Reply:
Or put it another way: as an user, I would expect `success` to be defined as *the TF resources are present* when I run the module with `state: present`. Even if I added `check_destroy: yes` (safety check). Anything less than that means that I, either manually or programmatically, need to perform other actions before I can have the TF resources present.

It may not be strictly a failure, but it will absolutely not be a success.

Reply:
Sounds reasonable to me.